astropy/io/fits/convenience.py

```diff
@@ -709,6 +710,105 @@ def info(filename, output=None, **kwargs):
     return ret
 
 
+def printdiff(inputa, inputb, *args, **kwargs):
+    """
+    Compare two parts of a fits file, including entire FITS files, FITS HDULists and
+    FITS HDU objects.
+
+    Parameters
+    ----------
+    inputa : str of file path, HDUList object, or HDU object
+        The filename of a FITS file on disk, or HDUList or HDU
+        object to compare to inputb.
+
+    inputb : str of file path, HDUList object, or HDU object
+        The filename of a FITS file on disk, or HDUList or HDU
+        object to compare to inputa.
+
+    ext, extname, extver
+        Additional positional arguments are for extension specification if your
+        inputA and inputB are string filenames (will not work if inputA and
+        inputB are HDU objects or HDULists). They are flexible and are best illustrated
+        by examples.  In addition to using these arguments positionaly you can
+        directly call the keyword parameters ext, extname.
+
+        By extension number::
+
+            printdiff('inA.fits', 'inB.fits', 0)      # the primary HDU
+            printdiff('inA.fits', 'inB.fits', 2)      # the second extension
+            printdiff('inA.fits', 'inB.fits', ext=2)  # the second extension
+
+        By name, i.e., ``EXTNAME`` value (if unique)::
+
+            printdiff('inA.fits', 'inB.fits', 'sci')
+            printdiff('inA.fits', 'inB.fits', extname='sci')  # equivalent
+
+        Note ``EXTNAME`` values are not case sensitive.
+
+        By combination of ``EXTNAME`` and ``EXTVER`` as separate
+        arguments or as a tuple::
+
+            printdiff('inA.fits', 'inB.fits', 'sci', 2)  # EXTNAME='SCI' & EXTVER=2
+            printdiff('inA.fits', 'inB.fits', extname='sci', extver=2)  # equivalent
+            printdiff('inA.fits', 'inB.fits', ('sci', 2))  # equivalent
+
+        Ambiguous or conflicting specifications will raise an exception::
+
+            printdiff('inA.fits', 'inB.fits', ext=('sci',1), extname='err', extver=2)
+
+    kwargs
+        Any additional keyword arguments to be passed to
+        `astropy.io.fits.FITSDiff`.
```

Comment:
In that case, I'm pretty neutral about changing this.  I can see how it would be nice for a newer user to see the full call, makes it less confusing when you're wanting to import said function that's referenced.  But I can also see @pllim point.